Text Scissoring Bug in Battle Mode Fix

authenticate opened the pull request:
Hi,

These changes should address the text clipping issue in battle mode, issue #528.

In addition, I included some other minor changes for the Windows build.

Let me know if you have any questions for concerns.

Thanks.

Bertram25 commented:
I'll cherry-pick your commits fixing the windows build to ease your overall Pull requests. :) Thanks for all the work done.

authenticate commented:
I force pushed up a new set of commits.  The scissor rectangle coordinates for the UI now go through the entire graphics pipeline.  I added a 4D vector class to help make the math more readable.

This should make the scissors rectangle resolution independent and correct the previous issues.

Finally, I also updated the dimensions of the UI widget.  Without the changes to "battle_command.cpp", the UI should look like this at all resolutions:

![vt_1](https://cloud.githubusercontent.com/assets/4543259/13518326/86c32546-e19a-11e5-960d-86e81bc0bac2.png)

The "B" in Bare-Hand gets clipped.  The updates to "battle_command.cpp" produce the following:

![vt_2](https://cloud.githubusercontent.com/assets/4543259/13518343/b60e2670-e19a-11e5-9b09-e0bd6bbfc723.png)

Regardless of which version of "battle_command.cpp" you end up using, the clipping should be resolution independent and always clip the same way now.

Thanks.

Bertram25 commented:
Works like a charm, thanks for the fast adaptation with vectors implementation as a bonus. :)
